crypto/bls/src/lib.rs

```diff
@@ -33,18 +33,70 @@ mod zeroize_hash;
 
 pub mod impls;
 
+use alloy_primitives::FixedBytes;
 pub use generic_public_key::{
     INFINITY_PUBLIC_KEY, PUBLIC_KEY_BYTES_LEN, PUBLIC_KEY_UNCOMPRESSED_BYTES_LEN,
 };
 pub use generic_secret_key::SECRET_KEY_BYTES_LEN;
 pub use generic_signature::{INFINITY_SIGNATURE, SIGNATURE_BYTES_LEN};
 pub use get_withdrawal_credentials::get_withdrawal_credentials;
+use safe_arith::SafeArith;
 pub use zeroize_hash::ZeroizeHash;
 
 #[cfg(feature = "supranational")]
 use blst::BLST_ERROR as BlstError;
 
-pub type Hash256 = ethereum_types::H256;
+pub type Hash256 = alloy_primitives::B256;
+pub trait FixedBytesExtended {
+    fn from_low_u64_be(value: u64) -> Self;
+    fn from_low_u64_le(value: u64) -> Self;
+    fn zero() -> Self;
+}
+
+impl<const N: usize> FixedBytesExtended for FixedBytes<N> {
```

Comment:
Rather than creating 3 copies of this trait, maybe we should pull it out into its own crate in `consensus` which is pulled in by `types`, `merkle_proof` and `bls`?

We could call the crate `fixed_bytes`

Reply:
Implemented in [5162316](https://github.com/sigp/lighthouse/pull/6078/commits/5162316d8927b69451837468c211c69576a1fbf3)